dev/reference/transactions/transaction-pessimistic.md

```diff
@@ -0,0 +1,98 @@
+---
+title: TiDB Pessimistic Transaction Mode
+summary: Learn about the pessimistic transaction mode of TiDB.
+category: reference
+---
+
+# TiDB Pessimistic Transaction Mode
+
+By default, TiDB implements an optimistic transaction mode, where the transaction commit might fail because of transaction conflicts. To make sure that the commit succeeds, you need to modify the application and add an automatic retry mechanism. You can avoid this issue by using the pessimistic transaction mode of TiDB.
+
+> **Warning:**
+>
+> The pessimistic transaction mode of TiDB is an **experimental** feature. It is *not recommended to apply it in the production environment*.
+
+## Behaviors of the pessimistic transaction mode
+
+Pessimistic transactions in TiDB behave similarly to those in MySQL. See the minor differences in [Known restrictions](#known-restrictions).
+
+- When you perform the `SELECT FOR UPDATE` statement, transactions read the last committed data and apply a pessimistic lock on the data being read.
+
+- When you perform the `UPDATE/DELETE/INSERT` statement, transactions read the last committed data to execute on them and apply a pessimistic lock on the modified data.
+
+- When a pessimistic lock is applied on a row of data, other write transactions attempting to modify the data are blocked and have to wait for the lock to be released.
+
+- When a pessimistic lock is applied on a row of data, other transactions attempting to read the data are not blocked and can read the committed data.
+
+- All the locks are released when the transaction is committed or rolled back.
+
+- Deadlocks in concurrent transactions can be detected by the deadlock detector. A DEADLOCK error which is the same as that in MySQL is returned.
+
+- TiDB supports both the optimistic transaction mode and pessimistic transaction mode in the same cluster. You can specify either mode for transaction execution.
+
+## Methods to enable the pessimistic transaction mode
+
+The pessimistic transaction mode is disabled by default because it is currently an experimental feature. Before enabling it, you need to add the following setting in the configuration file:
+
+```
+[pessimistic-txn]
+enable = true
+```
+
+When `enable` is set to `true`, the default transaction mode in TiDB is still optimistic. To enable the pessimistic transaction mode, choose any of the following methods that suits your needs:
+
+- Use the `BEGIN PESSIMISTIC;` statement to start the transaction in the pessimistic transaction mode. You can write it in comment style as `BEGIN /*!90000 PESSIMISTIC */;` to make it compatible with the MySQL syntax.
+
+- Execute the `set @@tidb_txn_mode = 'pessimistic';` statement to allow all the transactions processed in this session to be in the pessimistic transaction mode.
+
+- Enable the pessimistic transaction mode in the configuration file. This allows all transactions (except auto-committed single-statement ones) to adopt the pessimistic transaction mode.
+
+  ```
```

Comment:
Incorrect indentation. Please use four spaces. Now the display on the website is not what we want.